src/core/server/saved_objects/saved_objects_service.test.ts

```diff
@@ -171,14 +171,12 @@ describe('SavedObjectsService', () => {
       return expect(KibanaMigratorMock.mock.calls[0][0].callCluster()).resolves.toMatch('success');
     });
 
-    it('skips KibanaMigrator migrations when --optimize=true', async () => {
-      const coreContext = createCoreContext({
-        env: ({ cliArgs: { optimize: true }, packageInfo: { version: 'x.x.x' } } as unknown) as Env,
-      });
+    it('skips KibanaMigrator migrations when pluginsInitialized=false', async () => {
+      const coreContext = createCoreContext({ skipMigration: false });
       const soService = new SavedObjectsService(coreContext);
 
       await soService.setup(createSetupDeps());
-      await soService.start({});
+      await soService.start({ pluginsInitialized: false });
```

Comment:
Probably low impact, but we now have the `config` SO type that is registered by core, not plugins. The type would be usable but not migrated with PR changes.

Reply:
Probably the correct way to solve is to have the processes that need to run Core only run `setup` and extract whatever information they need.